src/App.tsx

```diff
@@ -1,14 +1,50 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
 import { peopleFromServer } from './data/people';
 
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timer = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timer);
+
+    timer = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [usedQuery, setUsedQuery] = useState('');
+
+  const timerOfChange = useCallback(debounce(setUsedQuery, 1000), []);
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    timerOfChange(event.target.value);
+  };
+
+  const filterPeople = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name.toLocaleLowerCase()
+        .includes(usedQuery.toLocaleLowerCase()));
+  }, [usedQuery]);
+
+  const onSelected = useCallback((newPerson: Person) => {
```

Comment:
```suggestion
  const handleSelect = useCallback((newPerson: Person) => {
```
a handler function should have `handle` prefix, the props name should start with `on` https://jaketrent.com/post/naming-event-handlers-react/